harness/determined/experimental/core_v2/_core_v2.py

```diff
@@ -166,7 +169,16 @@ def _init_context(
 
     # Unmanaged trials.
     # Construct the config.
-    checkpoint_storage = checkpoint_storage or config.checkpoint_storage
+    checkpoint_storage = (
+        checkpoint_storage
+        or config.checkpoint_storage
+        or str(pathlib.Path(appdirs.user_data_dir("determined")) / "checkpoints")
+    )
+    checkpoint_storage_dict: Dict[str, Any] = (
+        storage.shared._shortcut_to_config(checkpoint_storage, False)  # type: ignore
+        if type(checkpoint_storage) == str
+        else checkpoint_storage
+    )
```

Comment:
1. i'd rather not change `_shortcut_to_config` because this if on/off cluster logic is pretty specific to detached mode, at least for the time being (i.e. directory type can be used on cluster, too). we should put more thought into how to parse `file://` strings outside of detached mode. so for now could we just not use that util method and instantiate the dict object directly here?
2. i don't think we need to parse checkpoint strings into dicts here, 'cause it seems like `_make_v2_context` already does that down the line. 

so maybe this could be:
```python3
checkpoint_storage = checkpoint_storage or config.checkpoint_storage
if checkpoint_storage is None:
    checkpoint_storage = {
        "type": "directory",
        "container_path": str(pathlib.Path(appdirs.user_data_dir("determined")) / "checkpoints")
    }

```

seems a bit cleaner IMO

Reply:
I don't think we can do this because `checkpoint_storage or config.checkpoint_storage` can be string, and the checkpoint storage field in the [config](https://github.com/determined-ai/determined/blob/a2461f93e48d2c646c0fd948c8be5afc42930234/harness/determined/experimental/core_v2/_core_v2.py#L196) needs to be an object.